packages/elements/src/react/sign-in/choose-strategy.tsx

```diff
@@ -0,0 +1,144 @@
+'use client';
+
+import type { SignInFactor, SignInFirstFactor, SignInStrategy as TSignInStrategy } from '@clerk/types';
+import { Slot } from '@radix-ui/react-slot';
+import { useSelector } from '@xstate/react';
+import * as React from 'react';
+import type { ActorRefFrom } from 'xstate';
+
+import type { TSignInFirstFactorMachine } from '~/internals/machines/sign-in/machines';
+import { SignInRouterSystemId } from '~/internals/machines/sign-in/types';
+
+import { useActiveTags } from '../hooks';
+import { ActiveTagsMode } from '../hooks/use-active-tags.hook';
+import { SignInRouterCtx } from './context';
+
+// --------------------------------- HELPERS ---------------------------------
+
+const localStrategies: Set<TSignInStrategy> = new Set(['email_code', 'password', 'phone_code', 'email_link']);
+const resetPasswordStrategies: Set<TSignInStrategy> = new Set([
+  'reset_password_phone_code',
+  'reset_password_email_code',
+]);
+
+export function isResetPasswordStrategy(strategy: TSignInStrategy | null | undefined): boolean {
+  return !!strategy && resetPasswordStrategies.has(strategy as TSignInStrategy);
+}
+
+export function factorHasLocalStrategy(factor: SignInFactor | undefined | null): boolean {
+  if (!factor) return false;
+  return localStrategies.has(factor.strategy);
+}
+
+// --------------------------------- COMPONENTS ---------------------------------
+
+export type SignInVerificationsProps = WithChildrenProp<{ preferred?: TSignInStrategy }>;
+export type SignInChooseStrategyProps = WithChildrenProp;
+
+export function SignInChooseStrategy({ children }: SignInChooseStrategyProps) {
+  const routerRef = SignInRouterCtx.useActorRef();
+  const activeState = useActiveTags(routerRef, ['route:first-factor', 'route:choose-strategy'], ActiveTagsMode.all);
+
+  return activeState ? children : null;
+}
+
+const AVAILABLE_STRATEGY_NAME = 'SignInAvailableStrategy';
+
+export type SignInAvailableStrategyElement = React.ElementRef<'button'>;
+export type SignInAvailableStrategyProps = WithChildrenProp<{ asChild?: boolean; name: SignInFirstFactor['strategy'] }>;
+
+/**
+ * By default, renders a button which will trigger a change in the chosen sign-in strategy. It **must** be used within a `<Step name='choose-strategy'>` component.
```

Comment:
I guess we could add to the description that this component will only render something if the strategy is enabled with the respective `name`?